qiskit/circuit/operation.py

```diff
@@ -0,0 +1,50 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Quantum Operation Mixin."""
+
+from abc import ABC
+
+
+class Operation(ABC):
```

Comment:
It would be useful to add more description to this docstring about the `Operation` class.

Since it has no abstract methods is it something we want to allow instantiating directly? If not, it may be possible to raise in that situation using the `__new__` method to check for type.

Reply:
@ewinston Thanks! Agreed, an Operation should not be instantiated directly. However, being a newcomer to both Qiskit and python, I am somewhat struggling with making this work.

Here is what I've tried:
```
 def __new__(cls):
        if cls is Operation:
            raise CircuitError("An Operation Mixing should not be instantiated directly.")
        return super().__new__(cls)
```
However, this fails many tests.

Reply:
This passes test for me in python 3.8 when using tox and just has a couple lint warnings about unused arguments. What were you seeing?

Reply:
Thanks! I am using python 3.7, but I am not sure whether this makes the difference. The code above is currently commented out in Operation.py. When I uncomment it out and add the import line
`from qiskit.circuit.exceptions import CircuitError`, then I see multiple failing tests.
As as specific example:
`python -m unittest  test.python.circuit.test_gate_definitions.TestStandardGates.test_inverse_13`
leads to
```
  File "C:\Users\ALEXANDERIVRII\Desktop\Qiskit\qiskit-terra\test\python\circuit\test_gate_definitions.py", line 238, in test_inverse
    gate = gate_class(*float_vector)
TypeError: __init__() missing 2 required positional arguments: 'phi' and 'lam'
```
I can track the difference to `_get_free_params(CU3Gate)`, which returns `['theta', 'phi', 'lam']` without `__new__`, and which returns ['kwargs'] with `__new__`.

Reply:
@ewinston - you can see the failing tests (before commenting out the `__new__` method) in the CI here:
https://github.com/Qiskit/qiskit-terra/runs/4077907538?check_suite_focus=true